src/utils/storage/update_handlers/v0.19.3.ts

```diff
@@ -0,0 +1,48 @@
+import { DEFAULT_MAINNET_TOKENS } from '../../../constants/tokens';
+import { JSONWallet } from '../../../interfaces/plug_wallet';
+import { Mainnet, RegisteredToken } from '../../../PlugKeyRing/modules/NetworkModule/Network';
+
+export default (storage: any) => {
+  const defaultTokenIds = DEFAULT_MAINNET_TOKENS.map((t) => t.canisterId);
+  const wallets: JSONWallet[] = storage.wallets;
+
+  // For each wallet, get its tokens and add the wallet id to the list of registered by
+  const nestedTokens = wallets?.map((wallet) => (Object.values(wallet?.assets) || []).map(
+    (asset) => ({
+      ...(asset?.token || {}),
+      registeredBy: [wallet.walletNumber],
+      logo: asset?.token?.logo || (asset?.token as any)?.image,
+    }),
+    ));
+  const tokens = nestedTokens?.flat?.() || [];
+  // Remove default tokens and merge duplicates, concatting their registeredBy
+  const registeredTokens = tokens.reduce((acum, token) => {
+
+    // Remove default tokens
+    if (defaultTokenIds.includes(token.canisterId)) {
+      return acum;
+    }
+
+    // If we already had this token, only update the registeredby array
+    const registeredToken = acum.find((t) => t.canisterId === token.canisterId);
+    if (registeredToken) {
+      const updatedToken = { ...registeredToken, registeredBy: [...registeredToken.registeredBy, ...token.registeredBy] };
+      return acum.map((t) => t.canisterId === token.canisterId ? updatedToken : t);
+    }
```

Comment:
@rocky-fleek  Im not sure if this works, but what u think? So we don't map acum each time
```suggestion
    const registeredIndex = acum.findIndex((t) => t.canisterId === token.canisterId);
    if (registeredIndex >= 0) {
      const registeredToken = acum[registeredIndex];
      acum[registeredIndex] = { ...registeredToken, registeredBy: [...registeredToken.registeredBy, ...token.registeredBy] };
      return acum;
    }
```